src/proposed/0140-budget-limit-threshold.md

```diff
@@ -0,0 +1,469 @@
+<!--
+  SPDX-License-Identifier: CC-BY-SA-4.0
+  Copyright 2023 UNSW
+-->
+
+# MCS: Budget limit thresholds on endpoints for SC Donation
+
+- Author: Mitchell Johnston
+- Proposed: 2023-08-15
+
+## Summary
+
+This is a two stage RFC that proposes to add a new `budget threshold' to
+endpoints, which would only allow scheduling context donation to occur if the
+SC's available budget exceeds the threshold. This would provide a mechanism to
+prevent budget-expiry from occurring in passive servers.
+
+As a second optional stage, this threshold value could also restrict the budget
+that a passive server is permitted to consume on a donated SC. We refer to this
+as a `budget limit'. This would help bound priority inversion in a system as
+well as reducing the trust that clients must place in passive servers they call.
+
+## Motivation
+
+### Thresholds
+
+This change regarding thresholds is motivated by a desire to prevent budget
```

Comment:
suggestion: this would be better worded as "to provide an option for users to configure passive servers without requiring timeout handlers"